test/core/promise/pipe_test.cc

```diff
@@ -113,17 +132,26 @@ TEST(PipeTest, CanFilter) {
   EXPECT_CALL(on_done, Call(absl::OkStatus()));
   MakeActivity(
       [&pipe] {
+        // Setup some filters here, carefully getting ordering correct by doing
+        // so outside of the Join().
         auto doubler = pipe.receiver.Filter(
             [](int p) { return absl::StatusOr<int>(p * 2); });
         auto adder = pipe.sender.Filter(
```

Comment:
Could you explain why one filter is on the receiver side and the other on the sender side, as well as clarify what could happen if these were set up in the Join()?

Reply:
Done